src/video-interface.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+/**
+ * >>>>>>Work-In-Progress<<<<<<
+ *
+ * VideoInterface defines a common video API which any AMP component that plays
+ * videos is expected to implement.
+ *
+ * AMP runtime uses this common API to provide consistent video experience and
+ * analytics across all video players.
+ *
+ * Components implementing this interface must also extend
+ * {@link ./base-element.BaseElement} and register with the
+ * Video Manager {@link ./service/video-manager-impl.VideoManager} during
+ * their `builtCallback`.
+ *
+ * @interface
+ * @extends {./base-element.BaseElement}
+ */
+export class VideoInterface {
```

Comment:
I'd definitely also expect `getState()` of some sort here. E.g. "playing", "paused", "finished" or such.